lib/redacted_struct.rb

```diff
@@ -1,47 +1,41 @@
 # frozen_string_literal: true
 
-# A subclass of Struct that redacts members by default, and can allow some to be printed
-class RedactedStruct < Struct
-  VERSION = "1.1.0"
-
-  def self.new(*name_and_members, keyword_init: nil, allowed_members: [], &block)
-    super(*name_and_members, keyword_init: keyword_init, &block).tap do |struct_class|
-      struct_class.class_eval do
-        @allowed_members = Array(allowed_members)
-      end
-    end
-  end
-
-  class << self
-    attr_reader :allowed_members
-  end
+# Class methods for the Redactable module. When Redactable is included in class that class is
+# extended with these instance methods.
+module RedactableClassMethods
+  attr_reader :allowed_members
+end
 
-  def allowed_members
-    self.class.allowed_members
+# A module that adds redacted functionality to a class
+module Redactable
+  def self.included(mod)
+    mod.extend RedactableClassMethods
   end
 
   def inspect
     name_or_nil = self.class.name ? " #{self.class.name}" : nil
 
     attributes = members.map do |member|
       if allowed_members.include?(member)
-        "#{member}=#{self[member].inspect}"
+        "#{member}=#{send(member).inspect}"
```

Comment:
This is an unsafe send. I couldn't find a good way to avoid it without changing the API of Data. An alternative is to convert to a hash and call `#[]` on the hash.

Reply:
yeah maybe we convert it once to hash outside the loop?

```ruby
    members_h = to_h

    attributes = members.map do |member|
      if allowed_members.include?(member)
        "#{member}=#{members_h[member].inspect}"
```